tests/integration/test_dbapi_integration.py

```diff
@@ -19,6 +19,7 @@
 
 import trino
 from tests.integration.conftest import trino_version
+from trino import constants
```

Comment:
nit: `Refactor role into roles` -> `Add support for system and connector roles`?

Does it make sense (and is it accurate?)?

Reply:
It is really just a rename of role into roles. There is no new feature being introduced in that commit actually. The `role` field actually already contained multiple roles. I thought this is more consistent with the roles being passed on from `dbapi.connect`, introduced in the next commit.

Reply:
That's the reason why the first commit by itself fails tests. See 

```
======================================================================================================================================= FAILURES ========================================================================================================================================
_____________________________________________________________________________________________________________________ test_set_role_in_connection_trino_higher_351 ______________________________________________________________________________________________________________________

run_trino = (<Popen: returncode: None args: ['docker', 'run', '--rm', '-p', '56257:8080',...>, 'localhost', 56257)

    @pytest.mark.skipif(trino_version() == '351', reason="Newer Trino versions return the system role")
    def test_set_role_in_connection_trino_higher_351(run_trino):
        _, host, port = run_trino

>       trino_connection = trino.dbapi.Connection(
            host=host, port=port, user="test", catalog="tpch", roles={"system": "ALL"}
        )
E       TypeError: Connection.__init__() got an unexpected keyword argument 'roles'

tests/integration/test_dbapi_integration.py:1077: TypeError
------------------------------------------------------------------------------------------------------------------------------- Captured stdout teardown --------------------------------------------------------------------------------------------------------------------------------
trino-python-client-tests-637e3b1
____________________________________________________________________________________________________________________________ test_role_is_set_when_specified ____________________________________________________________________________________________________________________________

mock_client = <MagicMock name='client' id='4532039360'>

    @patch("trino.dbapi.trino.client")
    def test_role_is_set_when_specified(mock_client):
        roles = {"system": "finance"}
>       with connect("sample_trino_cluster:443", roles=roles) as conn:

tests/unit/test_dbapi.py:262:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = ('sample_trino_cluster:443',), kwargs = {'roles': {'system': 'finance'}}

    def connect(*args, **kwargs):
        """Constructor for creating a connection to the database.

        See class :py:class:`Connection` for arguments.

        :returns: a :py:class:`Connection` object.
        """
>       return Connection(*args, **kwargs)
E       TypeError: Connection.__init__() got an unexpected keyword argument 'roles'

trino/dbapi.py:82: TypeError
================================================================================================================================ short test summary info ================================================================================================================================
FAILED tests/integration/test_dbapi_integration.py::test_set_role_in_connection_trino_higher_351 - TypeError: Connection.__init__() got an unexpected keyword argument 'roles'
FAILED tests/unit/test_dbapi.py::test_role_is_set_when_specified - TypeError: Connection.__init__() got an unexpected keyword argument 'roles'
```

Reply:
I probably messed up something in my fixup comments locally. All good now.